KyberElasticAdapter/contracts/lib/Maintainable.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+/**
+ * @dev Contract module which extends the basic access control mechanism of Ownable
+ * to include many maintainers, whom only the owner (DEFAULT_ADMIN_ROLE) may add and
+ * remove.
+ *
+ * By default, the owner account will be the one that deploys the contract. This can
+ * later be changed with {transferOwnership}.
+ *
+ * This module is used through inheritance. It will make available this modifier:
+ * `onlyMaintainer`, which can be applied to your functions to restrict their use to
+ * the accounts with the role of maintainer.
+ */
+
+abstract contract Maintainable is Context, AccessControl {
+    bytes32 public constant MAINTAINER_ROLE = keccak256("MAINTAINER_ROLE");
+
+    constructor() {
+        address msgSender = _msgSender();
+        // members of the DEFAULT_ADMIN_ROLE alone may revoke and grant role membership
+        _setupRole(DEFAULT_ADMIN_ROLE, msgSender);
+        _setupRole(MAINTAINER_ROLE, msgSender);
+    }
+
+    function addMaintainer(address addedMaintainer) public virtual {
+        grantRole(MAINTAINER_ROLE, addedMaintainer);
+    }
+
+    function removeMaintainer(address removedMaintainer) public virtual {
```

Comment:
Missing NatSpec
Add descriptions for the interface, events and functions
Refer https://docs.soliditylang.org/en/v0.8.17/natspec-format.html